### PR TITLE
refactor(checkbox-input)!: convert to CSS modules

### PR DESCRIPTION
**Pull request by zchsh**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1199185162145258/f)
🔍 [Preview Link](https://react-components-git-zscheckbox-input-css-modules-hashicorp.vercel.app/components/checkboxinput)

---

This PR updates `@hashicorp/react-checkbox-input` to use CSS modules.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/checkbox-input/index.js [hunk 1]
```diff
@@ -1,45 +1,55 @@
...
       </div>
-      {error && <div className="g-type-body-small error">{error}</div>}
+      {error && <div className={s.errorText}>{error}</div>}
     </div>
   )
 }
 
-CheckboxInput.defaultProps = {
-  theme: { background: 'light' },
-}
```
Comment (zchsh) [inline packages/checkbox-input/index.js:41]: Moved this to a default value in props destructuring, as that seems to be a pattern we're strongly leaning towards in other work.

File: packages/checkbox-input/index.js [hunk 2]
```diff
@@ -1,45 +1,55 @@
...
+ */
+function CheckboxInput({
+  label,
+  theme = { background: 'light' },
+  field,
+  form,
+  className,
+}) {
+  //  `field` and `form` props typically come from Formik,
+  // though you can also provide them without Formik.
```
Comment (zchsh) [inline packages/checkbox-input/index.js:18]: Modified this comment a bit to clarify that this component is totally usable without Formik.

File: packages/checkbox-input/index.test.js [hunk 3]
```diff
@@ -9,11 +9,14 @@ const defaultProps = getTestValues(props)
...
-    const { container } = render(<CheckboxInput {...defaultProps} />)
+  it('should pass a provided className to the root element', () => {
+    const className = 'my-special-checkbox'
+    const { container } = render(
+      <CheckboxInput {...defaultProps} className={className} />
+    )
     const rootElem = container.firstChild
     expect(rootElem.tagName).toBe('DIV')
-    expect(rootElem).toHaveClass('g-checkbox-input')
+    expect(rootElem).toHaveClass(className)
```
Comment (zchsh) [inline packages/checkbox-input/index.test.js:19]: As mentioned in the `changeset`, and similar to many other component conversions to CSS modules, we no longer set  a global `g-{component-name}` class on the root element. However, to support [existing overrides](https://sourcegraph.hashi-mktg.com/search?q=g-checkbox-input&patternType=literal), we now support a `className` prop.

File: pages/global.css [hunk 4]
```diff
@@ -4,7 +4,6 @@
 
 /* these should be moved to css modules */
 @import '../packages/callouts/styles/index.css';
-@import '../packages/checkbox-input/style.css';
```
Comment (zchsh) [inline pages/global.css:7]: 🔪 🎉

**Review by BRKalow — APPROVED** [review]
🚢

**Pull request merged**